README.md

```
@@ -149,19 +149,34 @@ async for response in agent.run(
 
 Check out the [CSVAnalysisAgent documentation](https://kodeagent.readthedocs.io/en/latest/specialized_agents.html) for more details.
 
-### Memory(less)
+## 🧠 Memory and State Management
 
-By default, any agent in KodeAgent is **memoryless** across tasks—each task begins with no prior context, a clean slate. To enable context from the previous task (only), use **Recurrent Mode**:
+By default, any agent in KodeAgent is **memoryless** across tasks—each task begins with a clean slate. You have two ways to manage state across runs:
+
+#### 1. Recurrent Mode (Single-Task Context)
```

markdownlint-cli2 (0.22.1)

[warning] Heading levels should only increment by one level at a time
Expected: h3; Actual: h4

(MD001, heading-increment)